src/lib.rs

```diff
@@ -125,43 +250,80 @@ impl ElasticCollector {
 }
 
 impl Default for ElasticCollector {
-    fn default() -> ElasticCollector {
+    fn default() -> Self {
         ElasticCollector::new_local(IndexTemplate::default())
     }
 }
 
+fn build_batch(events: &[Event<'static>], template: &IndexTemplate) -> Vec<u8> {
+    let mut buf = Cursor::new(Vec::new());
+    let formatter = RenderedJsonFormatter::new();
+
+    for evt in events {
+        let idx = template.index(&evt.timestamp());
+
+        //Writes a header struct of the form: {"index":{"_index":"{}","_type":"{}"}}\n
+        buf.write_all(b"{\"index\":{\"_index\":\"").unwrap();
+        buf.write_all(idx.as_bytes()).unwrap();
+        buf.write_all(b"\",\"_type\":\"").unwrap();
+        buf.write_all(TYPENAME.as_bytes()).unwrap();
+        buf.write_all(b"\"}}\n").unwrap();
+
+        //Writes the message body to the buffer
+        formatter.write_event(&evt, &mut buf).unwrap();
+        buf.write(b"\n").unwrap();
+    }
```

Comment:
Does this result in a payload like:

```
<header>
<event>
<header>
<event>
...
```

If so, does Elasticsearch permit multiple event bodies per payload? Comparing `idx` with the previously-written index might get this down to a single header per batch, if so.